doc/ug/dv_methodology/index.md

```diff
@@ -465,7 +465,32 @@ It is recommended to follow these guidelines when collecting coverage:
 *  Collect all coverage metrics (except toggle based on above bullet) on the DUT and all of its non-pre-verified sub-modules
 *  Treat pre-verified sub-modules as ['black-box'](#integration-testing) in terms of coverage collection
 
-### X-Propagation (Xprop)
+### Unreachable Coverage Analysis
+
+Instead of manually reviewing coverage reports to find unreachable code, we use VCS UNR to generate a UNR exclusion file which lists all the unreachable codes.
+VCS UNR (Unreachability) is a formal solution that determines the unreachable coverage objects automatically from simulation.
+The coverage config file ([cover.cfg](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/cover.cfg)) used with `-cm_hier` in simulation will be used as UNR input to cater our [Coverage Collection Guidelines](#Coverage Collection Guidelines), and common exclusion file ([common_cov_excl.el](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/common_cov_excl.el)) can be also supplied to UNR to avoid generating reviewed common unreachable coverage.
+Follow these steps to run and submit the exclusion file.
+1. Generate a VCS coverage database for the block by using `--cov` option in regression tools-dvsim.py
+2. Use following command to run VCS UNR to generate exclusion file
+```
+util/dvsim/dvsim.py path/to/<dut>_sim_cfg.hjson --cov-unr
+```
+3. If no exclusion file is generated, there is no unreachable code in RTL.
+   If there is an exclusion file generated, the output should be reviewed by both designer and verification engineer.
+   When the unreachable code is sensible and we decide to exclude it in coverage report, create a PR to add to ['common_cov_excl.el'](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/common_cov_excl.el) or block specific exclusion file, such as ['uart_cov_excl.el'](https://github.com/lowRISC/opentitan/blob/master/hw/ip/uart/dv/cov/uart_cov_excl.el)
+
+Here are some guidelines for using UNR and checking in generating exclusion.
+1. It's encouraged that designers run UNR to check the design in the early design stage (D1/D2), but adding exclusion for unreachable coverage should be done between D3 and V3 stage when the design is frozen (no feature update is allowed, except bug fix).
+It is not the right thing to add a coverage exclusion file to reach the 80% needed for V2.
+2. If any RTL changes happen to the design after the coverage exclusion file has been created, it needs to be redone and re-reviewed.
+3. All coverage exclusion files and coverage configuration file (if it's not using default [cover.cfg](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/cover.cfg)) should be checked during sign-off.
+4. Keep the VCS generated `CHECKSUM` along with exclusions, which is served as a crosscheck to ensure that the exclusion isn't applied when there is some change on the corresponding codes and the exclusion is outdated.
```

Comment:
This means we'd have to forego the `--exclude-bypass-checks` switch.

Reply:
We don't use this switch except xbar. Xbar has a very few auto-generated exclusions, which we can review it during signoff.

Reply:
True, but worth mentioning this.